index.js

```diff
@@ -125,22 +127,27 @@ Carpenter.prototype.send = function send(pathname, options, next) {
     options.headers['Content-Type'] = 'application/json';
   }
 
-  //
-  // Setup hyperquest to formatted URL.
-  //
-  req = request(url.format(base), options, next);
-
-  //
-  // Write JSON data to the request if provided. Yhis is only required for
-  // POST and PUT as hyperquest by default ends GET, DELETE and HEAD requests.
-  //
-  try {
-    if (~methods.indexOf(options.method)) {
-      req.end(typeof data === 'object' ? JSON.stringify(data) : data);
+  var operation = new Retryme(this.retryOpts);
```

Comment:
so while ultimately we should just make this consistent with [`feedsme-api-client`](https://github.com/warehouseai/feedsme-api-client/blob/master/index.js#L108-L153), for now we need to preserve the streaming interface due to how its used [here](https://github.com/godaddy/warehouse.ai/blob/master/lib/npm/publisher.js#L326-L360). We have plans to change the interface carpenter responds with and when we do that we should make the breaking change here.